libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -828,6 +828,28 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				StorageByteArrayElement(m_context).storeValue(*type, _functionCall.location(), true);
 			break;
 		}
+		case FunctionType::Kind::ByteArrayPop:
+		case FunctionType::Kind::ArrayPop:
+		{
+			_functionCall.expression().accept(*this);
+			solAssert(function.parameterTypes().size() == 0, "");
+
+			// TODO: Extract proper type for regular arrays
+			shared_ptr<ArrayType> arrayType = make_shared<ArrayType>(DataLocation::Storage);
+
+			//get the current length
+			ArrayUtils(m_context).retrieveLength(*arrayType);
+			// stack: ArrayReference currentLength
+
+			// throw if table is too short
+			m_context << Instruction::DUP1 << Instruction::ISZERO;
+			m_context.appendConditionalInvalid();
+
+			m_context << u256(1) << Instruction::SUB;
+			// stack: ArrayReference newLength
+			ArrayUtils(m_context).resizeDynamicArray(*arrayType);
```

Comment:
This works for now, but can be optimized: It would be better to set the element to zero and then just reduce the length. This does not work for bytearrays, though.